src/Common/src/System/Security/Cryptography/Asn1V2.cs

```diff
@@ -0,0 +1,5246 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Security.Cryptography.Asn1
+{
+    // Uses a masked overlay of the tag class encoding.
+    // T-REC-X.690-201508 sec 8.1.2.2
+    internal enum TagClass : byte
+    {
+        Universal = 0,
+        Application = 0b0100_0000,
+        ContextSpecific = 0b1000_0000,
+        Private = 0b1100_0000,
+    }
+
+    internal enum UniversalTagNumber
+    {
+        EndOfContents = 0,
+        Boolean = 1,
+        Integer = 2,
+        BitString = 3,
+        OctetString = 4,
+        Null = 5,
+        ObjectIdentifier = 6,
+        ObjectDescriptor = 7,
+        External = 8,
+        InstanceOf = External,
+        Real = 9,
+        Enumerated = 10,
+        Embedded = 11,
+        UTF8String = 12,
+        RelativeObjectIdentifier = 13,
+        Time = 14,
+        // 15 is reserved
+        Sequence = 16,
+        SequenceOf = Sequence,
+        Set = 17,
+        SetOf = Set,
+        NumericString = 18,
+        PrintableString = 19,
+        TeletexString = 20,
+        T61String = TeletexString,
+        VideotexString = 21,
+        IA5String = 22,
+        UtcTime = 23,
+        GeneralizedTime = 24,
+        GraphicString = 25,
+        VisibleString = 26,
+        ISO646String = VisibleString,
+        GeneralString = 27,
+        UniversalString = 28,
+        UnrestrictedCharacterString = 29,
+        BMPString = 30,
+        Date = 31,
+        TimeOfDay = 32,
+        DateTime = 33,
+        Duration = 34,
+    }
+
+    internal enum NamedBitListMode
```

Comment:
Recommend being prescriptive and saying NamedZeroIsOne is our only use case. Callers can bit swap if they need compatibility with existing libraries. We can add a bit-swap utility function in the future if needed.